src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/WorkBranch.cs

```diff
@@ -33,10 +34,32 @@ public async Task MergeBackAsync(string commitMessage)
         _logger.LogInformation("Merging {branchName} into {mainBranch}", _workBranch, OriginalBranch);
 
         await _repo.CheckoutAsync(OriginalBranch);
-        var result = await _repo.ExecuteGitCommand(
-            [ "merge", _workBranch, "--no-commit", "--no-edit", "--squash", "-q" ]);
 
-        result.ThrowIfFailed($"Failed to merge work branch {_workBranch} into {OriginalBranch}");
+        var mergeArgs = new[] { "merge", _workBranch, "--no-commit", "--no-edit", "--squash", "-q" };
+
+        var result = await _repo.ExecuteGitCommand(mergeArgs);
+
+        // If we failed, it might be because the working tree is dirty because of EOL changes
+        // These are uninteresting because the index will have the correct EOLs
+        if (!result.Succeeded)
+        {
+            // Let's see if that is the case
+            var diffResult = await _repo.ExecuteGitCommand(["diff", "-w"]);
+            if (diffResult.Succeeded)
+            {
+                _logger.LogWarning("Update succeeded but some EOL file changes were not properly stashes. Ignoring these...");
+
+                // We stage those changes, they will get absorbed with upcoming merge
+                result = await _repo.ExecuteGitCommand(["add", "-A"]);
+                result.ThrowIfFailed($"Failed to stage whitespace-only EOL changes while merging work branch {_workBranch} into {OriginalBranch}");
+                result = await _repo.ExecuteGitCommand(mergeArgs);
+                result.ThrowIfFailed($"Failed to merge work branch {_workBranch} into {OriginalBranch}");
+            }
+            else
```

Comment:
we could also `ThrowIfFailed` instead of the else?
Not sure which one you prefer

Reply:
We could but maybe this is more explicit?